lib/diagnostic.js

```diff
@@ -5,17 +5,22 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
```

Comment:
I think you're confusing *implementing* reduce, which we did in class, and writing a callback that will be *passed* to reduce. (Which is much easier!)
Please review the reduce documentation on MDN.